types/src/transaction/automation.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024 Supra.
+
+use crate::transaction::user_transaction_context::EntryFunctionPayload;
+use crate::transaction::EntryFunction;
+use move_core_types::identifier::{IdentStr, Identifier};
+use move_core_types::language_storage::{ModuleId, CORE_CODE_ADDRESS};
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+
+struct AutomationTransactionEntryRef {
+    module_id: ModuleId,
+    entry_name: String,
+}
+
+static AUTOMATION_TRANSACTION_ENTRY: Lazy<AutomationTransactionEntryRef> =
+    Lazy::new(|| AutomationTransactionEntryRef {
+        module_id: ModuleId::new(
+            CORE_CODE_ADDRESS,
+            Identifier::new("automation_registry").unwrap(),
+        ),
+        entry_name: "register".to_string(),
+    });
+
+/// Represents options for Automation transaction payload.
+#[derive(Clone, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub enum AutomationTransactionPayload {
```

Comment:
Taking into account that we are doing already custom handling for the user transactions which aim automation task registration here we can have another option as well, instead of EntryFunction with call to  `automation_registry::register` we can have struct enclosing all arguments that are required to pass to the register function.
```
AutomationTransactionPayload {
        automated_txn_payload: EntryFunction,
        expiry_time: u64,
        max_gas_amount: u64,
        gas_price_cap: u64,
}
```
Then during transaction execution we can easily verify `automated_txn_payload` and then manually construct  a new entry-function to call `automation_registry::register` function with the above properties as arguments to the function.

With this approach this type of validity checked here https://github.com/Entropy-Foundation/aptos-core/pull/134/files#diff-fd22752e67147bf54afdcfa60f8074d0d6558f1610b34a940e6591d7bbdaec45R902 can be avoided.

Dr. @sjoshisupra what are your thoughts related to this.